### ISSUE TITLE
Bloomreach - Product ID and Category

### ISSUE THREAD

**Issue by jasonegan** [searchdiscovery/mohawk-flooring-dl-spec#100]
On a call today, Bloomreach indicated that they need specific formats for their product number (on product view, cart add and conversion/order completion) and category ID (on PLP).

These expected values that align to the Bloomreach feed need to be added to the data layer. From there, SDI will then update the Bloomreach pixels to capture these values.

**Comment by nate-wolfe** [comment]
For the sake of sanity, I'll name these in the events as brsm_pid and brsm_cat_id, within productInfo. 

From our conversation with Shreyas, we only need the PID in the add-to-cart and product viewed events, and the category ID on the PLP.

**Comment by jasonegan** [comment]
@nate-wolfe It looks like the product ID is also needed in the conversion pixel. Can you confirm with Shreyas that this does/does not need to be updated as well?
![image](https://user-images.githubusercontent.com/72522616/122615221-287d5780-d056-11eb-8b6d-a48d10863bed.png)

**Comment by nate-wolfe** [comment]
Since Order Placed builds heavily off of Checkout Started, I can only add the BRSM pid to Order Placed by also adding it to Checkout Started. Can we ignore the value in Checkout Started, since it isn't necessary?

**Comment by jasonegan** [comment]
@nate-wolfe Yes, you can add the variable to the data layer there as well, and we'll just not use it for anything. Thanks!

**Comment by jasonegan** [comment]
@rmessickSDI Can you make an update to the Bloomreach pixels to pull from the new product ID and cat ID vars here?

**Comment by rmessickSDI** [comment]
@nate-wolfe Do not see the cat in DL.  
https://staging.pergoflooring.com/shop/laminate/portfolio/detail/LF000967/rustic-smoked-chestnut?title=Pergo%20Portfolio%20+WetProtect

![image](https://user-images.githubusercontent.com/78565036/123150103-9a271e00-d427-11eb-91d5-9aae4df5781e.png)

**Comment by rmessickSDI** [comment]
![image](https://user-images.githubusercontent.com/78565036/123150518-13267580-d428-11eb-8033-5758f194adc9.png)
https://staging.pergoflooring.com/shop/hardwood?installation=&look=&page=1

**Comment by nate-wolfe** [comment]
@rmessickSDI I was under the impression that the category ID was needed on the PLP specifically, so it's in the Product Listing Displayed event. 
![image](https://user-images.githubusercontent.com/81365619/123151388-6650f600-d431-11eb-8895-477b37ed9110.png)
Is this incorrect? I can add it to Product Viewed without a problem, but I don't see how we would add it to Page Load Started. Category ID is Engineered, Laminate, or Luxury_Vinyl_Tile_And_Plank, so it corresponds to products rather than pages.

**Comment by rmessickSDI** [comment]
It is needed on the PLP page itself.  Bloomreach only wants the Cat for that page not per product.  I could look for just the first one but not sure if that is correct on all senerios.  

Also the documentation says this:
```
// INSERT the Category specific Tracking Parameters in your Global Tracking Script
  // in your CATEGORY PAGE TEMPLATE.
  br_data.cat_id = "106121";
  br_data.cat = "106120|106121";
 // End - Category page specific tracking parameters
```
 Is just cat_id all we need to map this field to?

**Comment by nate-wolfe** [comment]
That wouldn't be accurate in all scenarios. In the event they're on the shop all PLP, there will be multiple types. I'll still have to put it in the Product Listing Displayed event, but that fires every time the PLP loads. I can set the cat for the page outside of the product-specific data. 

I'm not sure the difference between cat and cat_id, though. If both are bound only to the page, not to individual listings, why would cat_id be a subset of cat?

**Comment by rmessickSDI** [comment]
@nate-wolfe @savithanuguri Can we get Shreyas feedback on this?  Can we have him tell us what should be set. for these pages and then we can work backward on how to get there.

https://staging.pergoflooring.com/shop/laminate?installation=&look=&page=1
https://staging.pergoflooring.com/shop/hardwood?installation=&look=&page=1
https://staging.pergoflooring.com/shop/rigid-vinyl?installation=&look=&page=1
https://staging.pergoflooring.com/shop?installation=&look=&page=1

Here is bloomreach docs on it:
https://documentation.bloomreach.com/developers/search-and-merchandising/pixel-deployment/page-view-pixels/category-page-view-pixel.html

**Comment by savithanuguri** [comment]
@rmessickSDI  and @nate-wolfe , we have to add in the page load started event of PLP. 
We are making changes to the code. Once deployed will update

**Comment by nate-wolfe** [comment]
In dev:
![image](https://user-images.githubusercontent.com/81365619/123285686-437b1c00-d4db-11eb-9166-226f0046ccd4.png)
This hasn't been pushed to staging, but I believe it's correct. Does this look good to you, @rmessickSDI ?

**Comment by rmessickSDI** [comment]
looks correct to me.  Let me know when it gets to stage and I can test.

**Comment by nate-wolfe** [comment]
In staging:
![image](https://user-images.githubusercontent.com/81365619/123313839-e2157600-d4f7-11eb-8e11-1fc191058bed.png)
@rmessickSDI

**Comment by rmessickSDI** [comment]
![image](https://user-images.githubusercontent.com/78565036/123642223-aa087e80-d7e8-11eb-851a-46b9e3bf6a9c.png)
@savithanuguri Do you want to follow up with Bloomreach to make sure all pixels are good?
